### PR TITLE
chat-headless: reset EventClient if active when restarting convo

### PR DESCRIPTION
**Pull request by popestr**
If a ChatEventClient is the active client when restarting the conversation, call its resetSession method and reset to the bot client.

TEST=manual

Ran with locally updated chat-core, saw expected results in test app. Added and updated unit tests to reflect new behavior.

### CODE REVIEW

File: packages/chat-headless/src/ChatHeadlessImpl.ts [hunk 1]
```diff
@@ -264,6 +264,21 @@ export class ChatHeadlessImpl implements ChatHeadless {
   }
 
   restartConversation() {
+    if (isChatEventClient(this.chatClient)) {
+      this.chatClient.resetSession();
+      let nextClient: ChatClient | undefined = undefined;
```
Comment (yen-tt) [inline packages/chat-headless/src/ChatHeadlessImpl.ts:269]: I think we should always set to bot client when restarting convo, instead of next client

File: packages/chat-headless/src/ChatHeadlessImpl.ts [hunk 2]
```diff
@@ -264,6 +266,11 @@ export class ChatHeadlessImpl implements ChatHeadless {
   }
 
   restartConversation() {
+    if (isChatEventClient(this.chatClient)) {
```
Comment (yen-tt) [inline packages/chat-headless/src/ChatHeadlessImpl.ts:269]: while this is not the case for our current amazon connect use case, technically bot/agent client can be an http or event client. I think we should check if this is actually an agent client, then set to botClient (and if agent client is of event client type then we can reset)

File: packages/chat-headless/src/models/ChatHeadless.ts [hunk 3]
```diff
@@ -110,7 +110,7 @@ export interface ChatHeadless {
    */
   initLocalStorage(): void;
   /**
-   * Resets all fields within {@link ConversationState}
+   * Resets all fields within {@link ConversationState} and the agent session, if one exists.
```
Comment (yen-tt) [inline packages/chat-headless/src/models/ChatHeadless.ts:113]: nit: lets update this to mention that we set the responder back to bot

File: packages/chat-headless/tests/chatheadless.test.ts [hunk 4]
```diff
@@ -28,7 +29,13 @@ const mockedMetaState: MetaState = {
 };
 
 beforeEach(() => {
-  jest.spyOn(coreLib, "provideChatCore").mockImplementation();
+  jest.spyOn(coreLib, "provideChatCore").mockImplementation(() => {
+    const client: ChatHttpClient = {
```
Comment (yen-tt) [inline packages/chat-headless/tests/chatheadless.test.ts:33]: is this change necessary?

Comment (popestr) [reply packages/chat-headless/tests/chatheadless.test.ts:33]: Yeah, since `resetConversation` now uses `isChatEventClient`, `provideChatCore` can no longer return `undefined`.

**Review by yen-tt — APPROVED** [review]

File: packages/chat-headless/package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@yext/chat-headless",
-  "version": "0.10.0",
+  "version": "0.11.0",
```
Comment (yen-tt) [inline packages/chat-headless/package.json:3]: this can be a patch version update, since it's a bug
